src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseTestCaseRepository.java

```diff
@@ -18,6 +20,34 @@
 
     Set<ProgrammingExerciseTestCase> findByExerciseId(Long exerciseId);
 
+    default ProgrammingExerciseTestCase findByIdWithExerciseElseThrow(Long testCaseId) {
+        return findByIdWithExercise(testCaseId).orElseThrow(() -> new EntityNotFoundException("Programming Exercise Test Case", testCaseId));
+    }
+
+    /**
+     * Returns the test case with the programming exercise
+     * @param testCaseId of the test case
+     * @return the test case with the programming exercise
+     */
+    @Query("""
+            SELECT tc FROM ProgrammingExerciseTestCase tc
+            LEFT JOIN FETCH tc.exercise ex
+            WHERE tc.id = :testCaseId
```

Comment:
```suggestion
            WHERE tc.id = :#{'testCaseId}
```

Reply:
What's the advantage of doing it this way?
Both options are used within Artemis and are afaik equal. I would stick to using the shorter solution unless the longer one has any advantage.

Reply:
@ole-ve Wrote "changed by mistake which is why I added the other references here as well.
https://github.com/ls1intum/Artemis/pull/4642#discussion_r805393112

Reply:
The code you are referring was existing before, and therefore was changed. This is new code.